appservice/src/createAppService/SiteRuntimeStep.ts

```diff
@@ -196,11 +196,18 @@ export class SiteRuntimeStep extends AzureWizardPromptStep<IAppServiceWizardCont
         ];
     }
 
-    private sortQuickPicksByRuntime(runtimeItems: IAzureQuickPickItem<ILinuxRuntimeStack>[], recommendedRuntimes: string[]): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
-        function getPriority(item: IAzureQuickPickItem<ILinuxRuntimeStack>): number {
-            const index: number = recommendedRuntimes.findIndex((runtime: string) => item.data.name.includes(runtime));
-            return index === -1 ? recommendedRuntimes.length : index;
-        }
-        return runtimeItems.sort((a: IAzureQuickPickItem<ILinuxRuntimeStack>, b: IAzureQuickPickItem<ILinuxRuntimeStack>) => getPriority(a) - getPriority(b));
+    private sortQuickPicksByRuntime(runtimeQuickPicks: IAzureQuickPickItem<ILinuxRuntimeStack>[], runtimeRecommendations: string[]): IAzureQuickPickItem<ILinuxRuntimeStack>[] {
+        return runtimeQuickPicks.sort((ti1: IAzureQuickPickItem<ILinuxRuntimeStack>, ti2: IAzureQuickPickItem<ILinuxRuntimeStack>) => {
+            const index1: number = runtimeRecommendations.findIndex((runtime: string) => ti1.data.name.includes(runtime.toLocaleLowerCase()));
+            const index2: number = runtimeRecommendations.findIndex((runtime: string) => ti2.data.name.includes(runtime.toLocaleLowerCase()));
+
+            if (index1 > -1) {
+                return -1;
+            } else if (index2 > -1) {
+                return 1;
+            } else {
+                return ti1.data.name.localeCompare(ti2.data.name);
```

Comment:
I don't think this will sort the runtimes for the recommended case. Try your changes with a node app open - I'm guessing the node runtimes won't be sorted. You need to check if `ti1` and `ti2` are the same runtime (aka both "node" or both "python") and if so sort by localCompare _first_. If they're not the same, that's when you use the `getPriority` function.

Reply:
Eh, it kinda sorts it?  But if we're not sorting it alphabetically, I'll probably do it your way

![image](https://user-images.githubusercontent.com/5290572/56325205-2e014f00-6126-11e9-8cef-739b8e89184f.png)

Reply:
**sigh**, I just noticed, but Java SE is listed twice because the display name for Java 8 and Java 11 are the same from the API.

I plan to add a description, either `java8` or `java11` to the runtimes that this is relevant too.  Is that an alright solution?

Reply:
![image](https://user-images.githubusercontent.com/5290572/56448756-ee259d80-62c6-11e9-83e1-347d59c8a5cd.png)

Reply:
I think ideally we only show the description when necessary (seems like java runtimes are the only ones that need it). But I'm fine with that screenshot